### PR TITLE
[Merged by Bors] - refactor: change argument order in `with` clause of `peel` tactic

### PR DESCRIPTION
**Pull request by j-loreaux**
Per this [Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/pulling.20continuity.20past.20integrals/near/411065361), this changes the argument order of `peel h with h_peel x y z ...` to `peel h with x y z ... h_peel`, as the former behavior was deemed confusing. 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/Peel.lean [hunk 1]
```diff
@@ -223,19 +223,19 @@ def peelArgsIff (l : List Name) : TacticM Unit := withMainContext do
       peelArgsIff hs
 
 elab_rules : tactic
-  | `(tactic| peel $[$num?:num]? $e:term $[with $n? $l?*]?) => withMainContext do
+  | `(tactic| peel $[$num?:num]? $e:term $[with $l?* $n]?) => withMainContext do
```
Comment (kmill) [inline Mathlib/Tactic/Peel.lean:226]: Could you leave the name as `n?` as a reminder it's an Option?

Comment (j-loreaux) [reply Mathlib/Tactic/Peel.lean:226]: @kmill it *shouldn't* be an option when `with` is present. This argument should be mandatory in that case. Also, I think that would make parsing impossible. How would you know whether the final name is the last variable name (in the case when this argument is not present) or the name of the peeled hypothesis (when it is present)?

Comment (kmill) [reply Mathlib/Tactic/Peel.lean:226]: The question mark is just part of the name, it doesn't affect the syntax pattern. I like using ? To mark when syntax pieces have an Option type. Correct me if I'm wrong about the type here.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**